### ISSUE TITLE
DataStore sync on Android physical devices takes very long time

### ISSUE THREAD

**Issue by szymondobrzanski** [aws-amplify/amplify-flutter#4678]
### Description

Greetings,

I've observed that syncing DataStore on Android physical devices running Android version 10 or higher is time-consuming, typically ranging from 20 seconds to 2 minutes. On devices with older Android versions (specifically tested with 7 and 8), the sync process is notably swifter, max 15 seconds. I haven't been able to replicate this issue on emulators. Comparatively, on iOS devices, the sync completes within a matter of seconds.

Upon initializing the app, I trigger the sync process to load all necessary data:

```
await Amplify.DataStore.start();
await Amplify.Hub.availableStreams[HubChannel.DataStore]!
        .firstWhere((event) => event.eventName == "ready");
```

Numerous Android users of my app have reported encountering the same issue. For now the only workaround I see is dropping DataStore and replacing it with API-based approach. However, this would force me to rewrite my application's infrastructure layer. While I've come across similar reported issues, they appear to have been closed without a definitive solution or explanation for the underlying cause.

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

Reproduce steps similar to case in my application:
1. Add 20+ models to schema.graphql
2. Populate them with data
3. Build app on android physical device with android version 10+
4. Register user
5. Start sync
6. Wait until sync finishes

### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.19.5

### Amplify Flutter Version

1.7.0

### Deployment Method

Amplify CLI

**Comment by Equartey** [comment]
Hi @szymondobrzanski, sorry you've ran into this issue. This is a known issue that we're working to resolve. We'll update you when we've released a fix.

**Comment by szymondobrzanski** [comment]
Hey @Equartey  @khatruong2009, do you have an ETA for this fix?

**Comment by Equartey** [comment]
@szymondobrzanski we do not have a concrete timeline, but I can assure you this is one of our top priorities. 

We have an open [PR #4279](https://github.com/aws-amplify/amplify-flutter/pull/4279) that should resolve this issue. 

We'll provide an update when its available.

**Comment by amitchaudhary140** [comment]
Yes this is much needed because my Android users are complaining a lot. Good to know that it is your top priority and we expect the fix to be launched soon.

**Comment by Equartey** [comment]
Hi @amitchaudhary140, our fix will only be for Amplify Flutter. If you are observing similar issues with an Amplify React Native project, please submit a issue to their repo: https://github.com/aws-amplify/amplify-js/issues/new/choose

**Comment by tanle17th** [comment]
Hi @Equartey, do you guys have a plan for shipping https://github.com/aws-amplify/amplify-flutter/pull/4279?

**Comment by NikaHsn** [comment]
Thank you for your patience. The issue has been resolved in version 1.8.0.

**Comment by NikaHsn** [comment]
Thank you for your patience. The issue has been fixed in version 1.8.0. and I'm closing it. However, if you encounter any issues after updating to version 1.8.0, please don't hesitate to reopen it.
